sonarts-core/src/rules/noAccessorFieldMismatchRule.ts

```diff
@@ -0,0 +1,139 @@
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-accessor-field-mismatch",
+    description: "Getters and setters should access the expected fields",
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-2870",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
+    const symbols = SymbolTableBuilder.build(sourceFile, program);
+    return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program, symbols));
+  }
+}
+
+class Walker extends tslint.ProgramAwareRuleWalker {
+
+  constructor(sourceFile: ts.SourceFile, options: tslint.IOptions, program: ts.Program,private readonly symbols: SymbolTable) {
+    super(sourceFile, options, program);
+  }
+
+  protected visitMethodDeclaration(method: ts.MethodDeclaration): void {
+    this.visitAccessor(method, this.setterOrGetter(method));
+    super.visitMethodDeclaration(method);
+  }
+
+  protected visitSetAccessor(accessor: ts.AccessorDeclaration): void {
+    this.visitAccessor(accessor, {type: "setter", name : Walker.getName(accessor)});
+    super.visitSetAccessor(accessor);
+  }
+
+  protected visitGetAccessor(accessor: ts.AccessorDeclaration): void {
+    this.visitAccessor(accessor, {type: "getter", name : Walker.getName(accessor)});
+    super.visitGetAccessor(accessor);
+  }
+
+  private visitAccessor(accessor: ts.MethodDeclaration | ts.AccessorDeclaration, setterOrGetter: {type: string, name: string} | undefined) {
+    if (!setterOrGetter) {
+      return;
+    }
+    const matchingField = Walker.matchingField(accessor, setterOrGetter.name);
+    if (Walker.isPublic(accessor) && accessor.body && matchingField && !this.fieldIsUsed(accessor, matchingField)) {
+      this.addFailureAtNode(accessor.name, `Refactor this ${setterOrGetter.type} so that it actually refers to the property '${matchingField.name.getText()}'`);
+    }
+  }
+
+  private static isPublic(method: ts.MethodDeclaration | ts.AccessorDeclaration) {
+    const modifier = nav.accessModifier(method);
+    return modifier && modifier.kind === ts.SyntaxKind.PublicKeyword;
+  }
+
+  private static getName(accessor: ts.AccessorDeclaration) {
+    return accessor.name.getText().toLowerCase();
+  }
+
+  private setterOrGetter(method: ts.MethodDeclaration | ts.AccessorDeclaration): {type: string, name: string} | undefined {
+    const name = method.name.getText().toLowerCase();
+    if (method.kind === ts.SyntaxKind.SetAccessor) {
+      return { type: "setter", name};
+    }
+    if (method.kind === ts.SyntaxKind.GetAccessor) {
+      return { type: "getter", name};
+    }
+    if (name.startsWith("set") || name.startsWith("Set")) {
+      if (method.parameters.length == 1) {
+        return { type: "setter", name: name.substring(3) };
+      }
+    }
+    if (name.startsWith("get") || name.startsWith("Get")) {
+      if (method.parameters.length == 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') == should be === [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440)